ext/opentelemetry-exporter-cloud-monitoring/src/opentelemetry/exporter/cloud_monitoring/__init__.py

```diff
@@ -0,0 +1,136 @@
+    MetricRecord,
+    MetricsExporter,
+    MetricsExportResult,
+)
+from opentelemetry.sdk.metrics.export.aggregate import CounterAggregator
+
+logger = logging.getLogger(__name__)
+MAX_BATCH_WRITE = 200
+
+
+# pylint is unable to resolve members of protobuf objects
+# pylint: disable=no-member
+class CloudMonitoringMetricsExporter(MetricsExporter):
+    """ Implementation of Metrics Exporter to Google Cloud Monitoring"""
+
+    def __init__(self, project_id=None, client=None):
+        self.client = client or MetricServiceClient()
+        if not project_id:
+            _, self.project_id = google.auth.default()
+        else:
+            self.project_id = project_id
+        self.project_name = self.client.project_path(project_id)
+        self._metric_descriptors = {}
+
+    def _add_resource_info(self, series: TimeSeries):
+        """ Add Google resource specific information (e.g. instance id, region)
+
+        :param series: ProtoBuf TimeSeries
+        :return:
+        """
+        # TODO: Leverage this better
+
+    def _batch_write(self, series: TimeSeries):
+        """ Cloud Monitoring allows writing up to 200 time series at once
+
+        :param series: ProtoBuf TimeSeries
+        :return:
+        """
+        write_ind = 0
+        while write_ind < len(series):
+            self.client.create_time_series(
+                self.project_name,
+                series[write_ind : write_ind + MAX_BATCH_WRITE],
+            )
+            write_ind += MAX_BATCH_WRITE
+
+    def _get_metric_descriptor(self, record: MetricRecord) -> MetricDescriptor:
+        """ We can map Metric to MetricDescriptor using Metric.name or
+        MetricDescriptor.type. We create the MetricDescriptor if it doesn't
+        exist already and cache it. Note that recreating MetricDescriptors is
+        a no-op if it already exists.
+
+        :param record:
+        :return:
+        """
+        descriptor_type = "custom.googleapis.com/OpenTelemetry/{}".format(
+            record.metric.name
+        )
+        if descriptor_type in self._metric_descriptors:
+            return self._metric_descriptors[descriptor_type]
+        descriptor = {
+            "name": "display_name",
+            "type": descriptor_type,
+            "display_name": record.metric.name,
+            "description": record.metric.description,
+            "labels": [],
+        }
+        for key, value in record.labels:
+            if isinstance(value, str):
+                descriptor["labels"].append(
+                    LabelDescriptor(key=key, value_type="STRING")
+                )
+            elif isinstance(value, int):
+                descriptor["labels"].append(
+                    LabelDescriptor(key=key, value_type="INT64")
+                )
+            elif isinstance(value, bool):
+                descriptor["labels"].append(
+                    LabelDescriptor(key=key, value_type="BOOL")
+                )
```

Comment:
Is it possible for the value to be something other than a `str`, `int`, or `bool`? (I assume that would be the result of someone misusing the API.) If so it might be good to log an error message to indicate that the label is being skipped.

Reply:
The API specifies only string values actually, but I'll add this check for API violation.